### ISSUE TITLE
Add a mechanism to add prefix/suffixes to configuration files

### ISSUE THREAD

**Issue by nirbheek** [mesonbuild/meson#1492]
Right now, with `configure_file()` if the `input:` kwarg is not set, we just dump the entire configuration. This has the advantage of not having to maintain a duplicate list of defines (one in `meson.build`, another in `config.h.in`) and is less error-prone.

However, this leaves a gap: sometimes the `config.h` is mostly generated, but has a few static defines in it. We should implement a way to also add these to the generated `config.h`. There are two ways to implement this:

1) Add `prefix:` and `suffix:` kwargs to `configure_file()`

2) Add a new token type `#meson_dump_configuration` (or similar) that can be put in `config.h.meson` that will dump the `configuration_data()` object in that place

The latter has the advantage of being more flexible, and the former has the advantage of having all the defines in one place (in `meson.build`).

**Comment by tp-m** [comment]
That still seems like a nice thing to have.

**Comment by dcbaker** [comment]
I agree. I think i like the second option better personally, but either would be nice

**Comment by eli-schwartz** [comment]
#6267 includes a proposal to add confdata.append(), which is sort of like the first suggestion here.
